dts/bindings/mipi-dbi/mipi-dbi-device.yaml

```diff
@@ -11,3 +11,9 @@ properties:
   mipi-max-frequency:
     type: int
     description: Maximum clock frequency of device's MIPI interface in Hz
+
+  mipi-mode:
+    type: int
+    description: |
+      MIPI DBI mode in use. See dt-bindings/mipi_dbi/mipi_dbi.h for a list
```

Comment:
Why not an enum?

Reply:
We could use an enum here- I didn't because I thought that MIPI DBI driver implementations would be easier to follow if the MIPI DBI bus types were defined in a header file the drivers could access and use.

We could also use an enum in devicetree and a header file for drivers, but then we'd need to keep both implementations in sync

Reply:
I don't like the description of a property in a DT binding say "go look at this header file in the zephyr tree for options of this property", ideally a user should be able to go to the [DT binding index](https://docs.zephyrproject.org/latest/build/dts/api/bindings.html) and see all the info about the binding including the values to set

also don't think that keeping a subsystem header and essential binding for said subsystem in sync would present a problem at all for someone familiar enough with the subsystem to be making changes to it

Reply:
Sure- that makes sense. The reason I want to have the values within a header file is to simply implementations for MIPI DBI drivers. Looking at the SPI driver binding, it seems the approach used there is to:
- define the setting as an enum, with a few fixed values
- duplicate the values in a DT binding header, and instruct the user to use those values. Drivers can also use this header.

Here's the SPI device binding as an example: https://github.com/zephyrproject-rtos/zephyr/blob/cd2c7874b95eed7505620534b4b216ef42b7ce2b/dts/bindings/spi/spi-device.yaml#L40

So for the MIPI DBI implementation, the DT binding would look something like this:
```
mipi-mode:
    type: enum
    description: |
      MIPI DBI mode in use. Use the macros not the actual enum value, here is the concordance
      list (see dt-bindings/mipi_dbi/mipi_dbi.h)
        0     MIPI_DBI_MODE_SPI_3WIRE
        1     MIPI_DBI_MODE_SPI_4WIRE
   enum:
     - 0
     - 1
```

Would this approach work for you? That way we have the values exposed in a header, but they are documented in the DT binding index as well

Reply:
that's fine

Reply:
Ok, updated the PR with this change